### PR TITLE
Add vulnerability scanning to Jenkins pipelines

### LINKED ISSUE #459: [New Feature]: Integrate Container Dependency Scan into CI Pipelines
### Checked for duplicates

Yes - I've already checked

### Alternatives considered

Yes - and alternatives don't suffice

### Related problems

As an extra security measure, the OPERA project wants to incorporate vulnerability scanning of our Docker container images via the open source tool [Grype](https://github.com/anchore/grype).

### Describe the feature request

For our purposes, the best place to integrate this step is the Build/Test and Release CI pipelines within our Jenkins instance. Grype is already installed to /usr/bin/local on the CI instance, so the main work for this ticket will be to add a step to run Grype and capture the results to file within the Jenkinsfile for the two pipelines. The results should then be archived with the job as a build artifact.

### PR DESCRIPTION
**Pull request by RKuttruff**
## Description
- This PR adds a Grype vulnerability scan to the Build/Test and Build/Release Jenkins pipelines.
- HTML reports containing only vulnerabilities with known fixes are added to each build.

## Affected Issues
- Resolves #459 

## Testing
- Aside from the Grype scan stage, testing remains unchanged
- Used sandbox pipeline to verify changes to both Jenkinsfiles

### CODE REVIEW

File: .ci/jenkins/html.tmpl
Comment (collinss-jpl) [inline .ci/jenkins/html.tmpl:1]: Lets give this file a slightly more descriptive name, something like `grype_html.tmpl`

File: .ci/jenkins/build-test-deploy/Jenkinsfile [hunk 1]
```diff
@@ -78,6 +78,27 @@ pipeline {
...
+            steps{
+                echo "Scanning ${DOCKER_IMAGE_PREFIX} docker images with tag ${DOCKER_TAG} for patchable vulnerabilities"
+                sh label: 'Update Grype DB',
+                   script: "grype db update"
+                script{
+                    DOCKER_IMAGE_SUFFIXES.tokenize(',').each { DOCKER_IMAGE_SUFFIX ->
+                        echo "Checking ${DOCKER_IMAGE_PREFIX}/${DOCKER_IMAGE_SUFFIX} docker image with tag ${DOCKER_TAG}"
+                        sh label: "Check ${DOCKER_IMAGE_PREFIX}/${DOCKER_IMAGE_SUFFIX} OPERA Docker image",
+                           script: """
+                           grype ${DOCKER_IMAGE_PREFIX}/${DOCKER_IMAGE_SUFFIX}:$DOCKER_TAG --only-fixed -o template -t .ci/jenkins/html.tmpl --file test_results/${DOCKER_IMAGE_SUFFIX}/vulnerabilities.html
```
Comment (collinss-jpl) [inline .ci/jenkins/build-test-deploy/Jenkinsfile:94]: Let's include grype in the name of the output html so its clear what the source is, something like `grype_vulnerabilities.html` or `grype_scan_results.html`

**Review by collinss-jpl — APPROVED** [review]
Everything looks great @RKuttruff, just had some suggestions on file names to make things a bit clearer.

File: .ci/jenkins/build-test-deploy/Jenkinsfile (hunk 1 above)
Comment (RKuttruff) [reply .ci/jenkins/build-test-deploy/Jenkinsfile]: Thanks Scott, made the suggested changes

**Pull request merged**